core/src/smartcontracts/isi/permissions/mod.rs

```diff
@@ -72,6 +139,78 @@ pub enum ValidatorVerdict {
     Allow,
 }
 
+/// Logical operations on [`ValidatorVerdict`] behave as follows:
+///
+/// | Not   |      |
+/// |-------|------|
+/// | Allow | Deny |
+/// | Skip  | Skip |
+/// | Deny  | Allow|
+///
+/// | And  | Allow | Skip  | Deny |
+/// |------|-------|-------|------|
+/// |Allow | Allow | Allow | Deny |
+/// |Skip  | Allow | Skip  | Deny |
+/// |Deny  | Deny  | Deny  | Deny |
+///
+///
+/// | Or   | Allow | Skip  | Deny  |
+/// |------|-------|-------|-------|
+/// |Allow | Allow | Allow | Allow |
+/// |Skip  | Allow | Skip  | Deny  |
+/// |Deny  | Allow | Deny  | Deny  |
+///
+impl PredicateSymbol for ValidatorVerdict {
+    #[allow(clippy::unnested_or_patterns)] // More verbose, but more readable here
+    fn and(self, other: Self) -> ControlFlow<Self, Self> {
+        match (self, other) {
```

Comment:
I think some branches could be condensed, but it's fine this way too.

Reply:
They could, I just find it more readable this way. I don't think it impacts performance much if at all.

Reply:
Ok, let's leave as is.